contracts/contracts/GrantRoundManager.sol

```diff
@@ -0,0 +1,167 @@
+
+  /// @notice Emitted when a new GrantRound contract is created
+  event GrantRoundCreated(address grantRound);
+
+  /// @notice Emitted when a donation has been made
+  event GrantDonation(
+    uint96 indexed grantId,
+    IERC20 indexed tokenIn,
+    uint256 amountIn,
+    uint256 amountOut,
+    GrantRound[] rounds
+  );
+
+  constructor(
+    GrantRegistry _registry,
+    ISwapRouter _router,
+    IERC20 _donationToken
+  ) {
+    require(_registry.grantCount() >= 0, "GrantRoundManager: Invalid registry");
+    require(address(_router).isContract(), "GrantRoundManager: Invalid router"); // Router interface doesn't have a state variable to check
+    require(_donationToken.totalSupply() > 0, "GrantRoundManager: Invalid token");
+
+    registry = _registry;
+    router = _router;
+    donationToken = _donationToken;
+  }
+
+  /**
+   * @notice Creates a new GrantRound
+   * @param _owner Grant round owner that has permission to update the metadata pointer
+   * @param _payoutAdmin Grant round administrator that has permission to payout the matching pool
+   * @param _registry Address that contains the grant metadata
+   * @param _startTime Unix timestamp of the start of the round
+   * @param _endTime Unix timestamp of the end of the round
+   * @param _metaPtr URL pointing to the grant round metadata
+   * @param _minContribution Miniumum donation amount that can be made using the given donation token
+   */
+  function createGrantRound(
+    address _owner,
+    address _payoutAdmin,
+    GrantRegistry _registry,
+    uint256 _startTime,
+    uint256 _endTime,
+    string memory _metaPtr,
+    uint256 _minContribution
+  ) external {
+    GrantRound _grantRound = new GrantRound(
+      _owner,
+      _payoutAdmin,
+      _registry,
+      donationToken,
+      _startTime,
+      _endTime,
+      _metaPtr,
+      _minContribution
+    );
+
+    emit GrantRoundCreated(address(_grantRound));
+  }
+
+  /**
+   * @notice Swap and donate to a grant
+   * @param  _donation Donation being made to a grant
+   */
+  function swapAndDonate(Donation calldata _donation) external payable {
+    // --- Validation ---
+    // Rounds must be specified
+    GrantRound[] calldata _rounds = _donation.rounds;
+    require(_rounds.length > 0, "GrantRoundManager: Must specify at least one round");
+
+    // Only allow value to be sent if the input token is WETH
+    IERC20 _tokenIn = _donation.tokenIn;
+    require(
+      (msg.value == 0 && address(_tokenIn) != WETH) || (msg.value > 0 && address(_tokenIn) == WETH),
+      "GrantRoundManager: Invalid token-value pairing"
+    );
+
+    // Ensure grant recieving donation exists in registry
+    uint96 _grantId = _donation.grantId;
+    require(_grantId < registry.grantCount(), "GrantRoundManager: Grant does not exist in registry");
+
+    // Iterate through each GrantRound to verify:
+    //   - The round has the same donationToken as the GrantRoundManager
```

Comment:
What is the reason we check to make sure the round has the same donationToken as the GrantRoundManager? If it's not taken in as a parameter it shouldn't be possible to not be the same.

Reply:
The round address is an input, so any round can be passed in. If the round was not created from this particular `GrantRoundManager` it may have a different token